app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,39 @@
+class VotesController < ApplicationController
+
+
+  def show
+  end
+
+  def new
+    @vote = Vote.new
+  end
+
+  def create
+    @work = Work.find_by(id: params[:work_id])
+    @user = User.find_by(id: session[:user_id])
+
+    if @user.nil?
+      flash[:error] = "Please log in to upvote media."
+      return
+      redirect_back(fallback_location: root_path)
+    else
+      @vote = Vote.new(user_id: session[:user_id], work_id: params[:work_id])
+    end
+
+    if @vote.save
+      flash[:success] = "Successfully upvoted!"
+      redirect_back(fallback_location: root_path)
+      return
+    else
+      flash[:error] = "A problem occurred: Could not upvote"
```

Comment:
Use the darn validation error!

```suggestion
      flash[:error] = @vote.errors.messages[:user_id].first
```